examples/design/EU-DEMO/EUDEMO_builders/tf_coils.py

```diff
@@ -0,0 +1,805 @@
+        build_config: Dict,
+        separatrix: Optional[BluemiraWire] = None,
+        keep_out_zone: Optional[BluemiraWire] = None,
+    ):
+        super().__init__(params, build_config)
+
+        self.parameterisation_cls: Type[
+            GeometryParameterisation
+        ] = get_class_from_module(
+            self.build_config["param_class"],
+            default_module="bluemira.geometry.parameterisations",
+        )
+
+        self.variables_map = self.build_config.get("variables_map", {})
+
+        self.file_path = self.build_config.get("file_path", None)
+
+        problem_class = self.build_config.get("problem_class", None)
+
+        if problem_class is not None:
+            self.problem_class = get_class_from_module(problem_class)
+            self.problem_settings = self.build_config.get("problem_settings", {})
+
+            self.opt_config = self.build_config.get("optimisation_settings", {})
+
+            self.algorithm_name = self.opt_config.get("algorithm_name", "SLSQP")
+            self.opt_conditions = self.opt_config.get("conditions", {"max_eval": 100})
+            self.opt_parameters = self.opt_config.get("parameters", {})
+
+        self.separatrix = separatrix
+        self.keep_out_zone = keep_out_zone
+
+    def _make_wp_xs(self, inboard_centroid: float) -> BluemiraWire:
+        """
+        Make the winding pack x-y cross-section wire (excluding insulation and
+        insertion gap)
+        """
+        d_xc = 0.5 * self.params.tk_tf_wp.value
+        d_yc = np.full(4, 0.5 * self.params.tk_tf_wp_y.value)
+        d_yc[:2] = -d_yc[:2]
+
+        x_c = np.full(4, inboard_centroid)
+        x_c[[0, -1]] -= d_xc
+        x_c[[1, 2]] += d_xc
+
+        wp_xs = make_polygon([x_c, d_yc, np.zeros(4)], closed=True)
+        return wp_xs
+
+    def _make_centreline_koz(self, keep_out_zone) -> BluemiraWire:
+        """
+        Make a keep-out-zone for the TF coil centreline optimisation problem.
+        """
+        # The keep-out zone is for the TF WP centreline, so we need to add to it to
+        # prevent clashes when the winding pack thickness and casing are added.
+        tk_offset = 0.5 * self.params.tf_wp_width.value
+        # Variable thickness of the casing is problematic...
+        # TODO: Improve this estimate (or use variable offset here too..)
+        tk_offset += 1.5 * self.params.tk_tf_front_ib.value
+        tk_offset += self.params.g_ts_tf.value
+        return offset_wire(keep_out_zone, tk_offset, open_wire=False, join="arc")
+
+    def _derive_shape_params(self, variables_map: Dict[str, str]) -> Dict:
+        shape_params = {}
+        for key, val in variables_map.items():
+            if isinstance(val, str):
+                val = getattr(self.params, val).value
+
+            if isinstance(val, dict):
+                if isinstance(val["value"], str):
+                    val["value"] = getattr(self.params, val["value"]).value
+            else:
+                val = {"value": val}
```

Comment:
Is `variables_map` being mutated here? Should it be?

Reply:
Unsure, changed it so it definitely isnt mutated